### PR TITLE
Trace GraphQL query executions

### PR DESCRIPTION
**Pull request by jamesdphillips**
## What is this change?

* Updates `graphql-go/graphql` module for it's "Extensions" feature. "Extensions" are in essence middleware that allow an implementer to wrap query executions and provide functionality, see: https://github.com/graphql-go/graphql/pull/448.
* Introduces a GraphQL middleware that captures the duration it took to perform different operations within a query execution. Parsing, validation, resolving field, and total resolver time are observed. Implementation borrows heavily from [Ruby implementation](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/tracing/prometheus_tracing.rb).
* Observations are written to a Prometheus summary.

## Why is this change necessary?

Will allow us to get detailed metrics on GraphQL query performance.

## Does your change need a Changelog entry?

Sure.

## Do you need clarification on anything?

No.

## Were there any complications while making this change?

Hubris.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

I don't believe any will be required.

## How did you verify this change?

Unit tests

### CODE REVIEW

File: graphql/middleware.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"context"
+
+	"github.com/graphql-go/graphql"
+)
+
+// Middleware is an interface for extending a service with operations that
+// wrap existing functionality.
+type Middleware interface {
+	// Init is used to help you initialize the extension
```
Comment (nikkixdev) [inline graphql/middleware.go]: Nitpick about comments being full sentences: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: graphql/tracing/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
+package tracing
+
+import (
+	"context"
+
+	time "github.com/echlebek/timeproxy"
+	"github.com/graphql-go/graphql"
+	"github.com/graphql-go/graphql/gqlerrors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sensu/sensu-go/util/strings"
```
Comment (nikkixdev) [inline graphql/tracing/prometheus.go]: Might want to use an alias here, as not to get confused with `strings` from the standard library.

```suggestion
	utilstrings "github.com/sensu/sensu-go/util/strings"
```

File: graphql/middleware.go (hunk 1 above)
Comment (jamesdphillips) [reply graphql/middleware.go]: addressed 🙏

File: graphql/tracing/prometheus.go (hunk 2 above)
Comment (jamesdphillips) [reply graphql/tracing/prometheus.go]: addressed 🙏

**Pull request merged**